string name
float64 x_center
float64 y_center
float64 x_grid
float64 y_grid
float64 z_grid
float64 distance
